src/frontend/src/btc/components/convert/ConvertToCkBTC.svelte

```diff
@@ -1,19 +1,80 @@
 <script lang="ts">
-	import { getContext } from 'svelte';
+	import { isNullish, nonNullish } from '@dfinity/utils';
+	import { setContext, getContext } from 'svelte';
 	import { BTC_MAINNET_TOKEN } from '$env/tokens.btc.env';
+	import IcCkListener from '$icp/components/core/IcCkListener.svelte';
+	import { loadAllCkBtcInfo } from '$icp/services/ckbtc.services';
+	import { initCkBTCMinterInfoWorker } from '$icp/services/worker.ck-minter-info.services';
+	import { ckBtcMinterInfoStore } from '$icp/stores/ckbtc.store';
+	import type { IcCkToken } from '$icp/types/ic-token';
+	import ConvertModal from '$lib/components/convert/ConvertModal.svelte';
 	import ButtonHero from '$lib/components/hero/ButtonHero.svelte';
 	import IconCkConvert from '$lib/components/icons/IconCkConvert.svelte';
+	import { authIdentity } from '$lib/derived/auth.derived';
 	import { isBusy } from '$lib/derived/busy.derived';
+	import { modalConvertBTCToCkBTC } from '$lib/derived/modal.derived';
+	import { tokens } from '$lib/derived/tokens.derived';
+	import {
+		initConvertContext,
+		CONVERT_CONTEXT_KEY,
+		type ConvertContext
+	} from '$lib/stores/convert.store';
 	import { HERO_CONTEXT_KEY, type HeroContext } from '$lib/stores/hero.store';
 	import { i18n } from '$lib/stores/i18n.store';
+	import { modalStore } from '$lib/stores/modal.store';
+	import { findTwinToken } from '$lib/utils/token.utils';
+
+	let ckBtcToken: IcCkToken | undefined;
+	$: ckBtcToken = findTwinToken({
+		tokenToPair: BTC_MAINNET_TOKEN,
+		tokens: $tokens
+	});
+
+	$: ckBtcToken,
+		(() =>
+			nonNullish(ckBtcToken) &&
+			setContext<ConvertContext>(
+				CONVERT_CONTEXT_KEY,
+				initConvertContext({
+					sourceToken: BTC_MAINNET_TOKEN,
+					destinationToken: ckBtcToken
+				})
+			))();
+
+	let minterInfoLoaded = false;
+
+	const openConvert = async () => {
+		if (isNullish(ckBtcToken)) {
+			return;
+		}
+
+		minterInfoLoaded = nonNullish($ckBtcMinterInfoStore?.[ckBtcToken.id]);
+
+		await loadAllCkBtcInfo({
```

Comment:
You on purpose does not display an animation while loading things?

We have a pattern for this, something like "Hold tight while we are loading ...." displayed over a backdrop until the modal gets opened

Reply:
Ah coolio